test/lib/pmd/PmdLanguageManager.test.ts

```diff
@@ -0,0 +1,66 @@
+import {expect} from 'chai';
+import {before} from 'mocha';
+import Sinon = require('sinon');
+import {Config, EngineConfigContent} from '../../../src/lib/util/Config';
+import * as PmdLanguageManager from '../../../src/lib/pmd/PmdLanguageManager';
+import {PmdEngine} from '../../../src/lib/pmd/PmdEngine';
+
+describe('PmdLanguageManager', () => {
+	describe('getSupportedLanguages()', () => {
+		const fakePmdConfig: EngineConfigContent = {
+			name: "pmd",
+			targetPatterns: [
+				"**/*.cls", "**/*.java", "**/*.js", "**/*.page", "**/*.component", "**/*.xml",
+				"!**/node_modules/**", "!**/*-meta.xml"
+			]
+		};
+		let finalConfig: EngineConfigContent = null;
+
+		before(() => {
+			Sinon.createSandbox();
+			// getEngineConfig() should return the fake config that we've spoofed.
+			Sinon.stub(Config.prototype, 'getEngineConfig').withArgs(PmdEngine.NAME).returns(fakePmdConfig);
+			// setEngineConfig() should assign the config to finalConfig so we can see what it's trying to set.
+			Sinon.stub(Config.prototype, 'setEngineConfig').callsFake(async (name: string, ecc: EngineConfigContent) => {
+				finalConfig = ecc;
+			});
+		});
+
+		after(() => {
+			Sinon.restore();
+			finalConfig = null;
+		});
+
+		describe('When Config DOES specify supported languages', () => {
+			it('Explicitly supported languages are returned, and config is NOT updated', async () => {
+				// Simulate a config that specifies Apex, JavaScript, and Java.
+				fakePmdConfig.supportedLanguages = ['apex', 'javascript', 'java'];
+
+				const langs = await PmdLanguageManager.getSupportedLanguages();
+				expect(langs).to.deep.equal(['apex', 'javascript', 'java'], 'Explicitly supported languages should have been returned');
```

Comment:
I like your practice of including a meaningful error message with failures. Will adapt this for my new tests as well.